app/models/post.rb

```diff
@@ -1,3 +1,12 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', counter_cache: :post_counter
+  has_many :like
+  has_many :comment
+  def increment_post_counter
+    author.increment!(:post_counter)
+  end
```

Comment:
- [ ] Good job working on this method, but it will be better if you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#increment_post_counter` method every time a new post has been saved to database, then make the `increment_post_counter` method private.

Hint:

```ruby
  after_save :increment_post_counter

  private

  def increment_post_counter
    author.increment!(:posts_counter)
  end
```

Reply:
Done